File: Android/FirstAndroidApp/android/src/main/java/com/hashem/android/view/MainActivity.java
package com.hashem.android.view;

import android.os.Bundle;
import android.util.SparseArray;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.hashem.android.R;

import java.util.Optional;
import java.util.concurrent.CompletableFuture;
import java.util.stream.IntStream;

/**
 * Created by Hashem.
 */

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        IntStream.range(5, 10).forEach(System.out::println);

        SparseArray<String> sparseArray = new SparseArray<>();

        Optional<String> optional = Optional.ofNullable(null);

        CompletableFuture<Void> awdawd = CompletableFuture.runAsync(() -> {
            System.out.println("CompletableFuture");
        });

        awdawd.thenRun(() -> {
            System.out.println("CompletableFuture thenRun");
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        System.out.println("onDestroy");
    }

    @Override
    protected void finalize() throws Throwable {
        super.finalize();
        System.out.println("finalize");
    }
}

File: Android/FirstAndroidApp/android/src/main/java/com/hashem/android/view/MainActivity.kt
package com.hashem.android.view

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.hashem.android.MySingleton
import com.hashem.android.R

/**
 * Created by Hashem.
 */

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        MySingleton.getInstance {
            findViewById<TextView>(R.id.btn).text = "Hi from Singleton"
        }

        findViewById<TextView>(R.id.btn).setOnClickListener { v: View? ->
            startActivity(Intent(this, MainActivity::class.java))
            finish()
        }

        Runtime.getRuntime().gc()

        Bundle().apply {
            putString("key", "value")
        }
    }
}
